kroxylicious-docs/docs/_modules/record-encryption/azure-key-vault/proc-azure-key-vault-setup-microsoft-identity-platform-oauth2.adoc

```diff
@@ -0,0 +1,59 @@
+:_mod-docs-content-type: PROCEDURE
+
+// file included in the following:
+//
+// con-azure-key-vault-setup.adoc
+
+[id='proc-azure-key-vault-setup-microsoft-identity-platform-oauth2-{context}']
+
+= Authenticating with Microsoft Identity Platform via OAuth 2.0
+
+[role="_abstract"]
+This procedure describes how to create a Service Principal (an application identity) within Microsoft Entra ID to allow the Record Encryption filter to authenticate to Azure Key Vault using the OAuth 2.0 Client Credentials flow. The process uses the Azure CLI to create the application identity and assign the required Azure Role-Based Access Control (RBAC) role.
+
+NOTE: This identity is used solely for application-to-application (machine-to-machine) authentication. Ensure the Service Principal is only granted the minimal permissions required (the required access is defined below) as excessive access would unnecessarily increase the security risk.
```

Comment:
```suggestion
NOTE: This identity is used solely for application-to-application (machine-to-machine) authentication. Grant the Service Principal only the minimum permissions needed to access Azure Key Vault to avoid increasing security risk.
```